src/main/java/ro/developmentfactory/arrayProblem/IntermediateSuma.java

```diff
@@ -0,0 +1,35 @@
+package ro.developmentfactory.arrayProblem;
+
+
+import lombok.Getter;
+
+import static java.util.Arrays.copyOfRange;
+
+@Getter
+public class IntermediateSuma implements Runnable {
+    private static final int[] ab = { 23, 45, 67, 89, 12, 34, 56, 78, 90, 1 };
+    private static final int size = ab.length;
+    private  int totalSum ;
+
+    @Override
+    public void run() {
+        arrayDivide();
+    }
+
+    private void arrayDivide() {
+        int[] arrayFirstPart = copyOfRange(ab, 0, size / 2);
+        int[] arraySecondPart = copyOfRange(ab, size / 2, size);
+
+        int sumFirstPart = getElementSum(arrayFirstPart);
+        int sumSecondPart = getElementSum(arraySecondPart);
+        this.totalSum = sumFirstPart + sumSecondPart;
```

Comment:
with the current code, there is no need to separate the array in 2 parts - you are doing all calculations in 1 thread anyway